sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -0,0 +1,552 @@
+            resource_location = body.get("resourceLocation")
+            if resource_location:
+                return resource_location
+
+        if self.request.method in {"PUT", "PATCH"}:
+            return self.request.url
+
+        if self.request.method == "POST" and self.location_url:
+            return self.location_url
+
+        return None
+
+    def set_initial_status(self, pipeline_response):
+        # type: (PipelineResponseType) -> str
+        """Process first response after initiating long running operation.
+
+        :param azure.core.pipeline.PipelineResponse response: initial REST call response.
+        """
+        self.request = pipeline_response.http_response.request
+        response = pipeline_response.http_response
+
+        self._set_async_url_if_present(response)
+
+        if response.status_code in {200, 201, 202, 204} and self.async_url:
+            return "InProgress"
+        raise OperationFailed("Operation failed or canceled")
+
+    def _set_async_url_if_present(self, response):
+        # type: (ResponseType) -> None
+        async_url = response.headers.get(self._operation_location_header)
+        if async_url:
+            self.async_url = async_url
+
+        location_url = response.headers.get("location")
+        if location_url:
+            self.location_url = location_url
+
+    def get_status(self, pipeline_response):
+        # type: (PipelineResponseType) -> str
+        """Process the latest status update retrieved from an "Operation-Location" header.
+
+        :param azure.core.pipeline.PipelineResponse response: The response to extract the status.
+        :raises: BadResponse if response has no body, or body does not contain status.
+        """
+        response = pipeline_response.http_response
+        if _is_empty(response):
+            raise BadResponse(
+                "The response from long running operation does not contain a body."
+            )
+
+        body = _as_json(response)
+        status = body.get("status")
+        if not status:
+            raise BadResponse("No status found in body")
+        return status
+
+
+class LocationPolling(LongRunningOperation):
+    """Implements a Location polling.
+    """
+
+    def __init__(self):
+        self.location_url = None
+
+    def can_poll(self, pipeline_response):
+        # type: (PipelineResponseType) -> bool
+        """Answer if this polling method could be used.
+        """
+        response = pipeline_response.http_response
+        return "location" in response.headers
+
+    def get_polling_url(self):
+        """Return the polling URL.
+        """
+        return self.location_url
+
+    def get_final_get_url(self, pipeline_response):
+        # type: (PipelineResponseType) -> Optional[str]
+        """If a final GET is needed, returns the URL.
+
+        :rtype: str
+        """
+        return None
+
+    def set_initial_status(self, pipeline_response):
```

Comment:
Why this is a read-only method but the name is "set"?

Reply:
This is setting internal fields, and act like the "initialize" of the LROPoller. Not sure I want to rename it.

Reply:
Sorry I missed "self._set_async_url_if_present(response)"